app/models/chat_channel.rb

```diff
@@ -31,9 +31,9 @@ def add(user)
         UserChatChannelMembership.find_or_initialize_by(user_id: user.id, chat_channel: self)
 
       if !membership.following
+        update!(user_count: (user_count || 0) + 1)
```

Comment:
This line and the change in the file is to ensure we have updated user count before generating the membership object which will be used in the serializer, that way we get the correct count from the channel, as the membership is pulling it from here